src/v3/src/util/formUtils.ts

```diff
@@ -37,3 +54,215 @@ export const handleFormFieldBlur = (
+) : IdxRemediation[] => {
+  const widgetRemediations = transaction.neededToProceed;
+  // @ts-expect-error OKTA-609461 - idps missing from WidgetProps type
+  const idpsConfig = widgetProps.idps;
+
+  if (Array.isArray(idpsConfig)) {
+    const existsRedirectIdpIds: Record<string, boolean> = {};
+    widgetRemediations.forEach((r) => {
+      if (r.name === IDX_STEP.REDIRECT_IDP && r.idp) {
+        existsRedirectIdpIds[r.idp.id] = true;
+      }
+    });
+    const { baseUrl } = widgetProps;
+    const { stateHandle } = transaction.context;
+    return idpsConfig
+      .filter((c) => !existsRedirectIdpIds[c.id]) // omit idps that are already in remediation.
+      .map((idpConfig) => {
+        const idp = {
+          id: idpConfig.id,
+          name: idpConfig.text,
+        };
+        const redirectUri = `${baseUrl}/sso/idps/${idpConfig.id}?stateToken=${stateHandle}`;
+        if (idpConfig.className) {
+          // @ts-expect-error OKTA-609464 - className missing from IdpConfig type
+          idp.className = idpConfig.className;
+        }
+        return {
+          name: IDX_STEP.REDIRECT_IDP,
+          type: idpConfig.type,
+          idp,
+          href: redirectUri,
+        };
+      });
+  }
+
+  return [];
+};
+
+// TO DO: OKTA-609477 - Add rest of supported social idp icons here
+const getIdpButtonIcon = (idpType: string) : string => {
+  const idpIconMapping: Record<string, string> = {
+    facebook: facebookIconSvg,
+    google: googleIconSvg,
+    linkedin: linkedinIconSvg,
+    microsoft: msIconSvg,
+    apple: appleIconSvg,
+    okta: oktaIconSvg,
+  };
+  return idpIconMapping[idpType];
+};
+
+/**
+ * Example of `redirect-idp` remediation.
+ * {
+ *   "name": "redirect-idp",
+ *   "type": "MICROSOFT",
+ *   "idp": {
+ *      "id": "0oa2szc1K1YPgz1pe0g4",
+ *      "name": "Microsoft IDP"
+ *    },
+ *   "href": "http://localhost:3000/sso/idps/0oa2szc1K1YPgz1pe0g4?stateToken=BB...AA",
+ *   "method": "GET"
+ * }
+ *
+ */
+export const getIdpButtonElements = (
+  transaction: IdxTransaction,
+  widgetProps: WidgetProps,
+) : ButtonElement[] => {
+  const { neededToProceed: remediations } = transaction;
+  let redirectIdpRemediations = remediations.filter(
+    (idp) => idp.name === IDX_STEP.REDIRECT_IDP,
+  ) || [];
+  // Add Idp buttons from widget config (openid-connect)
+  redirectIdpRemediations = redirectIdpRemediations.concat(
+    getConfigIdpButtonRemediations(transaction, widgetProps),
+  );
+
+  // create button elements from idp objects
+  const idpButtonElements = redirectIdpRemediations.map((idpObject) => {
+    let type = idpObject.type?.toLowerCase() || '';
+    let displayName;
+
+    if (IDP.SUPPORTED_SOCIAL_IDPS.includes(type) === false) {
+      type = 'general-idp';
+      // OKTA-396684 - makes sure that custom idps always have a name
+      displayName = loc('customauth.sign.in.with.label', 'login', [idpObject.idp?.name]);
+    } else {
+      displayName = loc(`socialauth.${type}.label`, 'login');
```

Comment:
I believe we try to avoid using variables when referencing translation keys, the reason for this is because I think we eventually want to enforce that the key exists (possibly using typescript?) and using a variable like this will prevent that. 

What we have done in the past is put the string in a Record/map and based on the key (which can be accessed by passing it the variable) passing that value to the `loc` function . See [this](https://github.com/okta/okta-signin-widget/blob/0.5/src/v3/src/constants/idxConstants.ts#L213-L218) for reference